### PR TITLE
ENH: initial dummy types plugin

### PR DESCRIPTION
**Pull request by jairideout**
IntSequence1, IntSequence2, and Mapping types ported from QIIME 2 framework dummy plugin (`qiime.core.testing`). This package closely matches the q2-types package structure.

### CODE REVIEW

File: q2_dummy_types/plugin_setup.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+import q2_dummy_types
+
+plugin = qiime.plugin.Plugin(
+    name='dummy-types',
+    version=q2_dummy_types.__version__,
+    website='https://github.com/qiime2/q2-dummy-types',
+    package='q2_dummy_types'
+)
+
```
Comment (ebolyen) [inline q2_dummy_types/plugin_setup.py:21]: If this is illustrative, could you add a comment explaining the importance of importing these modules in this file?

File: q2_dummy_types/_int_sequence.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            return True
+
+int_sequence_data_layout = qiime.plugin.DataLayout('int-sequence', 1)
+int_sequence_data_layout.register_file('ints.txt', IntSequenceFormat)
+
+plugin.register_data_layout(int_sequence_data_layout)
+plugin.register_type_to_data_layout(IntSequence1, 'int-sequence', 1)
+plugin.register_type_to_data_layout(IntSequence2, 'int-sequence', 1)
+
+
```
Comment (ebolyen) [inline q2_dummy_types/_int_sequence.py:68]: Could you comment these general sections explaining what the registrations are accomplishing? e.g. We are associating a reader / writer for a given data layout to a particular object view.

**Comment by ebolyen** [comment]
Looks great, I think some comments would be useful for this package.

**Comment by jairideout** [comment]
Good idea, I added several descriptive comments to the important parts of the plugin.

**Comment by ebolyen** [comment]
:+1: :+1: :+1:

**Pull request merged**